docs/motivation-and-use-cases.md

```diff
@@ -0,0 +1,55 @@
+ipfs-cluster is a tool to coordinate ipfs nodes together.  It was clearly described here during its early planning stages: https://github.com/ipfs/notes/issues/58.
+From this description ipfs-cluster can be used to coordinate any aspect of an ipfs node’s state among a group of ipfs nodes.  This includes pin sets, the current focus of the project, but could also include state related to bitswap strategies or inter-node trust models, etc.  To achieve coordination of complex state among ipfs nodes without centralization, the ipfs-cluster project relies on a consensus protocol.  Today this is raft but in the future we want to make this be pluggable.  Today the project is solely focused on coordinating pinset state: https://github.com/ipfs/ipfs-cluster/blob/master/README.md.  This is a sensible place to focus because ipfs is at its core a protocol for storing and moving data and the pinset of an ipfs node determines what data it stores and distributes.
+
+
+## A key concept: the ipfs vNode abstraction:
+Early discussion, again in https://github.com/ipfs/notes/issues/58, outlines a particular approach that remains relevant to discussion today: ipfs-cluster as a virtual ipfs node (vNode for short).  The idea is that ipfs-cluster nodes could implement the ipfs api, only exposing state agreed upon by all nodes through consensus.  A simple example: all nodes in the cluster agree on a single peer id, and after reaching agreement all nodes respond with this id to requests on their ipfs vNode id endpoint.  A more useful example: an ipfs-cluster node gets an `add <file>` request through the ipfs vNode add endpoint, the cluster nodes coordinate adding this file, perhaps doing something clever like replicating file data across multiple nodes, and reach agreement that the data was added and pinned successfully via consensus.  After all this occurs the api endpoint returns the normal message indicating success.
+
+Designing ipfs-cluster to work this way has some benefits, including a familiar api for user interaction, the ability to use a cluster anywhere an ipfs node is used and the ability to make ipfs-clusters depend on other ipfs-clusters as the ipfs nodes that they coordinate.  This last property has the potential to make scaling ipfs-cluster easier; if large groups of participants can be abstracted away consensus peer group size can remain bounded as cluster participants grow arbitrarily.  It is not always the case that an ipfs api is the best user interface for adding files to ipfs-cluster.  If ipfs-cluster were to support behavior like per-pin replication configuration information, for example different pins specifying different replication factors as it does today, then the ipfs api has no endpoint to encode this information and some kind of cluster specific interface is needed  (See discussion https://botbot.me/freenode/ipfs/2017-02-09/ here for a somewhat related conversation that includes discussion of more use cases).
+
```

Comment:
We do support 'per-pin' replication configuration information. You can assign a different replication factor to every pin.